public/app/plugins/datasource/loki/language_provider.ts

```diff
@@ -54,7 +60,7 @@ export default class LokiLanguageProvider extends LanguageProvider {
   started: boolean;
   initialRange: AbsoluteTimeRange;
 
-  constructor(datasource: any, initialValues?: any) {
+  constructor(datasource: DataSourceApi, initialValues?: any) {
```

Comment:
I assume there is nothing else that can fit here except for `DataSourceApi` on the other hand we need `metadataRequest` there which actually isn't defined on `DataSourceApi` interface. So not ideal situation and I left `LanguageProvider` as it is with `this.datasource: any` (that is why `this.datasource.metadataRequest` call does not complain). Seems like I already added a lots of things to this PR, but we should probably make this more typesafe later.